pkg/api/types.go

```diff
@@ -165,6 +171,13 @@ const (
 	TerminationMessagePathDefault string = "/dev/termination-log"
 )
 
+// ClusterReference contains information to identify a cluster.
+type ClusterReference struct {
+	// The name of the cluster.
+	// Cannot be updated.
```

Comment:
I think @bgrant0607 needs to approve this change, since it affects every object in the system.